### PR TITLE
TST: xfail flatiter test for s390x and numpy<1.22

### LINKED ISSUE #11804: TST: TestQuantityReshapeFuncs.test_flat_attributes failed with AssertionError on s390x (big-endian)
Example log: https://github.com/astropy/astropy/runs/2708622607?check_suite_focus=true

Last week's run was successful (https://github.com/astropy/astropy/runs/2653579480?check_suite_focus=true) on 2021-05-24.

```
________________ TestQuantityReshapeFuncs.test_flat_attributes _________________

self = <astropy.units.tests.test_quantity_array_methods.TestQuantityReshapeFuncs object at 0x42b00b16a0>

    def test_flat_attributes(self):
        """While ``flat`` doesn't make a copy, it changes the shape."""
        q = np.arange(6.).reshape(3, 1, 2) * u.m
        qf = q.flat
        # see TestQuantityArrayCopy.test_flat for tests of iteration
        # and slicing and setting. Here we test the properties and methods to
        # match `numpy.ndarray.flatiter`
        assert qf.base is q
        # testing the indices -- flat and full -- into the array
        assert qf.coords == (0, 0, 0)  # to start
        assert qf.index == 0
        # now consume the iterator
        endindices = [(qf.index, qf.coords) for x in qf][-2]  # next() oversteps
>       assert endindices[0] == 5
E       assert 0 == 5

astropy/units/tests/test_quantity_array_methods.py:138: AssertionError
```

Could this be related to #11796 , @mhvk or @nstarman ?

Also see #11783, numpy/numpy#19153

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to fix #11804 .

A direct follow-up of #11796. Also see numpy/numpy#19153.

p.s. Ignore the `linkcheck` failure.

### CODE REVIEW

**Review by mhvk — APPROVED** [review]
Looks good, thanks!

**Comment by pllim** [comment]
Thanks for the approval! We'll find out in a few hours. 😸

**Comment by mhvk** [comment]
Darn, it the `xfail` didn't work - it ran the test and complained it failed:
```
self = <astropy.units.tests.test_quantity_array_methods.TestQuantityReshapeFuncs object at 0x415d6d9128>

    @pytest.mark.xfail(IS_S390X and NUMPY_LT_1_22, reason="Numpy GitHub Issue 19153")
    def test_flat_attributes(self):
        """While ``flat`` doesn't make a copy, it changes the shape."""
        q = np.arange(6.).reshape(3, 1, 2) * u.m
        qf = q.flat
        # see TestQuantityArrayCopy.test_flat for tests of iteration
        # and slicing and setting. Here we test the properties and methods to
        # match `numpy.ndarray.flatiter`
        assert qf.base is q
        # testing the indices -- flat and full -- into the array
        assert qf.coords == (0, 0, 0)  # to start
        assert qf.index == 0
        # now consume the iterator
        endindices = [(qf.index, qf.coords) for x in qf][-2]  # next() oversteps
>       assert endindices[0] == 5
E       assert 0 == 5

astropy/units/tests/test_quantity_array_methods.py:143: AssertionError
```

File: astropy/units/tests/test_quantity_array_methods.py [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+import os
 
 import pytest
 import numpy as np
 from numpy.testing import assert_array_equal
 
 from astropy import units as u
+from astropy.utils.compat import NUMPY_LT_1_22
+
+IS_S390X = os.environ.get('ARCH_ON_CI', 'normal') == 's390x'
```
Comment (mhvk) [inline astropy/units/tests/test_quantity_array_methods.py:12]: Must be this line...

Comment (pllim) [reply astropy/units/tests/test_quantity_array_methods.py:12]: Why is the `xfail` not working? 🤯 

`ARCH_ON_CI: s390x`
`Numpy: 1.20.3`

**Comment by mhvk** [comment]
If we're sure it fails because it is big-endian, one could do
```
xfail(sys.byteorder == 'big' and NUMPY_LT_1_22)
```

**Comment by pllim** [comment]
@mhvk , good idea. Let's try that but have to wait a few more hours. 😿

**Comment by pllim** [comment]
@mhvk , finally green! Should I squash and `[ci skip]` or should we just merge?

**Comment by mhvk** [comment]
Let's go for it! Thanks again!

**Pull request merged**